src/containerapp/azext_containerapp/_up_utils.py

```diff
@@ -882,26 +1092,76 @@ def _get_dockerfile_content(repo, branch, token, source, context_path, dockerfil
 
 
 def _get_app_env_and_group(
-    cmd, name, resource_group: "ResourceGroup", env: "ContainerAppEnvironment", location
+    cmd, name, resource_group: "ResourceGroup", env: "ContainerAppEnvironment", location, custom_location: "CustomLocation"
 ):
     if not resource_group.name and not resource_group.exists:
         matched_apps = [c for c in list_containerapp(cmd) if c["name"].lower() == name.lower()]
         if env.name:
             matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["environmentId"])["name"].lower() == env.name.lower()]
+        if env.custom_location_id:
+            matched_apps = [c for c in matched_apps if (c.get("extendedLocation") and c["extendedLocation"]["name"].lower() == env.custom_location_id.lower())]
+        elif env.resource_type:
+            matched_apps = [c for c in matched_apps if parse_resource_id(c["properties"]["environmentId"])["resource_type"].lower() == env.resource_type.lower()]
+        if custom_location.connected_cluster_id:
+            matched_apps = _filter_containerapps_by_connected_cluster_id(cmd, matched_apps, custom_location.connected_cluster_id)
         if location:
             matched_apps = [c for c in matched_apps if format_location(c["location"]) == format_location(location)]
         if len(matched_apps) == 1:
             resource_group.name = parse_resource_id(matched_apps[0]["id"])[
                 "resource_group"
             ]
             env.set_name(matched_apps[0]["properties"]["environmentId"])
+            if env.is_connected_environment():
+                env.custom_location_id = matched_apps[0]["extendedLocation"]["name"]
         elif len(matched_apps) > 1:
             raise ValidationError(
                 f"There are multiple containerapps with name {name} on the subscription. "
                 "Please specify which resource group your Containerapp is in."
             )
 
 
+def _filter_containerapps_by_connected_cluster_id(cmd, matched_apps, connected_cluster_id):
+    result_apps = []
+    for app in matched_apps:
+        if app.get("extendedLocation"):
+            custom_location_from_app = get_custom_location(cmd=cmd, custom_location_id=app["extendedLocation"]["name"])
+            if custom_location_from_app and connected_cluster_id.lower() == custom_location_from_app.host_resource_id.lower():
+                result_apps.append(app)
+    return result_apps
+
+
+# If the connected cluster already have a custom location (with ext namespace) bind to the container app ext, uses it.
+def _prepare_custom_location_args(
+        cmd,
+        custom_location: "CustomLocation",
+        extension: "Extension"
+):
+    if custom_location.name is None:
+        if custom_location.connected_cluster_id is None:
+            raise ValidationError("please specify one of connected-cluster-id or custom location you want to create the Connected environment.")
```

Comment:
customer also could create with --environment, right

Reply:
The `_prepare_custom_location_args` is for scenario without `--environment`
        if not env.check_exists():
            _prepare_custom_location_args(cmd, custom_location=custom_location, extension=extension)

I will update the message to:
`please specify one of --connected-cluster-id or --custom-location you want to create the Connected Environment or specify the existing Connected Environment with --environment.`